packages/cli/src/lib/snapshot/snapshot.spec.ts

```diff
@@ -0,0 +1,84 @@
+jest.mock('../platform/authenticatedClient');
+jest.mock('fs');
+
+import {createReadStream, ReadStream} from 'fs';
+import {join} from 'path';
+import {Readable} from 'stream';
+import {mocked} from 'ts-jest/utils';
+import {AuthenticatedClient} from '../platform/authenticatedClient';
+import {SnapshotFactory} from './snapshotFactory';
+
+const mockedCreateReadStream = mocked(createReadStream);
+const mockedAuthenticatedClient = mocked(AuthenticatedClient);
+const mockedCreateSnapshotFromFile = jest.fn();
+
+const doMockReadStream = () => {
+  mockedCreateReadStream.mockImplementation(() => {
+    const buffer = Buffer.from('this is a tést');
+    const readable = new Readable();
+    readable._read = () => {};
+    readable.push(buffer);
+    readable.push(null);
+    return readable as unknown as ReadStream;
```

Comment:
You probably could do something like `createReadStream(Buffer.from('this is a tést'))`

You can use the callback of mockImplementation to alter it properly:
https://jestjs.io/docs/mock-function-api#mockfnmockimplementationfn

Reply:
Not sure I understand how I can mock the createReadStream by using it. Do you have an example?